src/components/DeleteTableDialog.vue

```diff
@@ -193,15 +193,23 @@ export default Vue.extend({
         workspace,
       } = this;
 
-      const graphNames = await api.graphs(workspace);
-
+      const graphNames = (await api.graphs(workspace)).results.map((graph) => graph.name);
       const using = [] as Array<{graph: string; tables: string[]}>;
       graphNames.forEach(async (graph) => {
-        const data = await api.graph(workspace, graph);
+        const nodes = await api.nodes(workspace, graph, {});
+        // eslint-disable-next-line no-underscore-dangle
+        const edges = await api.edges(workspace, graph, nodes.results[0]._id, {
+          direction: 'all',
+        });
 
+        // eslint-disable-next-line no-underscore-dangle
+        const prelimNodes = nodes.results.map((node) => node._id.substring(0, node._id.indexOf('/')));
+        const nodeTables = prelimNodes.filter((node, index) => prelimNodes.indexOf(node) === index);
+        // eslint-disable-next-line no-underscore-dangle
+        const edgeTable = edges.results.length > 0 ? edges.results[0]._id.substring(0, edges.results[0]._id.indexOf('/')) : '';
         const tables: string[] = [];
         selection.forEach((table) => {
-          if (table === data.edgeTable || data.nodeTables.indexOf(table) > -1) {
+          if (table === edgeTable || nodeTables.indexOf(table) > -1) {
```

Comment:
Check out the [`Array.includes()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) function--it's probably better here than `indexOf()`.

Reply:
See https://github.com/multinet-app/multinet-client/commit/ceedc898a945927e4e277d13583c9eec195e6c64#diff-59d96b3b10e2aff2a38e84b28bc951f257462a7772757d307d3952647372bd4dL212-R217